internal/backend/runtime/omni/controllers/omni/machine_status_test.go

```diff
@@ -408,11 +397,50 @@ func (suite *MachineStatusSuite) TestMachineSchematic() {
 				KernelArgs:       kernelArgs,
 			},
 		},
+		{
+			name: "agent mode empty list",
+			extensions: []*runtime.ExtensionStatusSpec{
+				{
+					Metadata: extensions.Metadata{
+						Name:        constants.SchematicIDExtensionName,
+						Description: "0",
+						Version:     "full-id",
+					},
+				},
+				{
+					Metadata: extensions.Metadata{
+						Name:        "metal-agent",
+						Description: "1",
+					},
+				},
+				{
+					Metadata: extensions.Metadata{
+						Name:        "hello-world-service",
+						Description: "2",
+					},
+				},
+			},
+			expected: &specs.MachineStatusSpec_Schematic{
+				Id:               defaultSchematic,
+				InitialSchematic: defaultSchematic,
+				FullId:           defaultSchematic,
+			},
+		},
 	} {
 		suite.T().Run(tt.name, func(t *testing.T) {
 			ctx, cancel := context.WithTimeout(suite.ctx, time.Second*5)
 			defer cancel()
 
+			id := "test-" + tt.name
+
+			machine := omni.NewMachine(resources.DefaultNamespace, id)
+			spec := machine.TypedSpec().Value
+
+			spec.Connected = true
+			spec.ManagementAddress = suite.socketConnectionString
+
+			suite.Require().NoError(suite.state.Create(suite.ctx, machine))
+
```

Comment:
Moved this here to create a different machine ID each time - otherwise `MachineStatus` preserves the initial schematic, and my new test was giving different results when it was run in isolation vs when it was run together with other tests.